pkg/apis/maistra/v1/helmvalues.go

```diff
@@ -0,0 +1,134 @@
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// HelmValues is typedef for Helm .Values
+// +kubebuilder:validation:XPreserveUnknownFields
+type HelmValues struct {
+	data map[string]interface{} `json:"-"`
+}
+
+func NewHelmValues(values map[string]interface{}) *HelmValues {
+	return &HelmValues{data: values}
+}
+
+func (h *HelmValues) GetContent() map[string]interface{} {
+	if h == nil {
+		return nil
+	}
+	return h.data
+}
+
+func (h *HelmValues) GetField(path string) (interface{}, bool, error) {
+	if h == nil || h.data == nil {
+		return nil, false, nil
+	}
+
+	var val interface{} = h.data
+	fields := strings.Split(path, ".")
+	for i, field := range fields {
+		if m, ok := val.(map[string]interface{}); ok {
+			val, ok = m[field]
+			if !ok {
+				return nil, false, nil
+			}
+		} else {
+			return nil, false, fmt.Errorf("%v accessor error: %v is of the type %T, expected map[string]interface{}", strings.Join(fields[:i+1], "."), val, val)
+		}
+	}
+	return val, true, nil
+}
+
+func (h *HelmValues) GetBool(path string) (bool, bool, error) {
```

Comment:
Would it be better to use the code in `unstructured` for this?  Saves us having to maintain similar code.

Reply:
You mean use `unstructured` instead of `HelmValues`? Or just call `unstructured.NestedBool` in the `HelmValues' GetBool` function? That might be doable. I wanted to simplify stuff, but I see I can keep the simplifications, while still calling the existing unstructured code.

Reply:
I was thinking just implement `HelmValues.GetBool()` using the unstructured helpers.

Reply:
Fixed. Thanks!